R/stan_betareg.fit.R

```diff
@@ -0,0 +1,225 @@
+# Part of the rstanarm package for estimating model parameters
+# Copyright (C) 2013, 2014, 2015, 2016 Trustees of Columbia University
+# 
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 3
+# of the License, or (at your option) any later version.
+# 
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+#' @export
+#' @rdname stan_betareg
+
+stan_betareg.fit <- function (x, y, z, weights = rep(1, NROW(x)), offset = rep(0, NROW(x)),
+                              link = c("logit", "probit", "cloglog", "cauchit", "log", "loglog"), 
+                              link.phi = c("log", "identity", "sqrt"), ...,
+                              prior = normal(), prior_intercept = normal(),
+                              prior_z = normal(), prior_intercept_z = normal(),
+                              prior_ops = prior_options(), prior_PD = FALSE, 
+                              algorithm = c("sampling", "optimizing", "meanfield", "fullrank"),
+                              adapt_delta = NULL, QR = FALSE, sparse = FALSE) {
+  
+  # lots of tedious but simple stuff including standata which is a big list to pass to data {}
+  # process the prior information like stan_glm.fit() does
+  
+  algorithm <- match.arg(algorithm)
+
+  # determine whether the user has passed a matrix for the percision model (z)
+  if (length(link.phi) == 1) {
```

Comment:
Additionally, what happens if a user omits the `z` argument when calling `stan_betareg.fit` directly? Will it work? Without trying it my hunch is that it will error somewhere. That means that the user needs to specify something for `z` even if they don't have it in their model?